vllm/v1/core/sched/scheduler.py

```diff
@@ -296,9 +295,20 @@ def schedule(self) -> SchedulerOutput:
                     # is always a multiple of the block size. This limitation
                     # can potentially be removed in the future to slightly
                     # improve the performance.
-                    num_computed_tokens -= self.block_size
-                    num_new_tokens = self.block_size
-                    computed_blocks.pop()
+                    kv_cache_groups = self.kv_cache_config.kv_cache_groups
+                    if len(kv_cache_groups) > 1 or \
+                        not isinstance(kv_cache_groups[0].kv_cache_spec,
+                                    FullAttentionSpec):
+                        # It is difficult to handle the last block problem
+                        # for hybrid models. Ignore all computed tokens as
+                        # a temporary solution.
```

Comment:
Is it only the case for hybrid models, or is it the same for SWA-only models?

Reply:
Also, I don't quite understand what the fundamental problem is. Could you please elaborate more?